distributed/dashboard/components/scheduler.py

```diff
@@ -904,7 +905,15 @@ def update(self):
             if first_end > self.last:
                 last = self.last
                 self.last = first_end
-                if first_end > last + self.clear_interval * 1000:
+
+                # gap to previous is larger than clear_multiplier * average of currently visualized
+                # and greater than 10s
+                if len(self.source.data["start"]) > 2:
+                    gap_to_previous = self.source.data["start"][-1] - self.source.data["start"][-2] + self.source.data["duration"][-2]
+                else:
+                    gap_to_previous = 0
+                avg_duration_of_visualized = np.mean(self.source.data["duration"])
```

Comment:
I wonder if we might be able to avoid some of the cost in these checks by keeping track of a last-time-we-got-a-rectangle measurement.

Reply:
yes, I could add that, reducing the check only to once per second rather than calculating the mean each time a square is added. Although It would need an enormous amount of squares to cause the scheduler dashboard to slow down.

Reply:
Added a check that the np.mean will only be called at maximum once every 5 seconds. Otherwise, no reset will happen and the check is not performed.

Reply:
Any views of anything else needs to be added? Or any other concerns?